.github/workflows/upgrade-tests-reusable.yaml

```diff
@@ -0,0 +1,106 @@
+name: Upgrade Tests Reusable Workflow
+
+env:
+  E2E_LOG_LEVEL: debug
+  KYMA_STABILITY: "unstable"
+  KYMA: "./hack/kyma"
+
+on:
+  workflow_call:
+    inputs:
+      pre-upgrade-image-tag:
+        required: true
+        type: string
+      post-upgrade-image-tag:
+        required: true
+        type: string
+      build-job-name:
+        type: string
+      build-job-commit-sha:
+        type: string
+
+jobs:
+  e2e-upgrade: # This job tests the upgrade of Eventing module from latest image of the main branch to the current commit.
+    runs-on: ubuntu-latest
+    env:
+      BACKEND_TYPE: NATS
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Pre-upgrade checkout
+        run: git checkout -b ${{ inputs.pre-upgrade-image-tag }}
+
+      - name: Install k3d tools
+        run: make -C hack/ci/ install-k3d-tools
+
+      - name: Install Kyma CLI & setup k3d cluster using kyma CLI
+        run: |
+          make kyma
+          make -C hack/ci/ create-k3d
+          kubectl version
+          kubectl cluster-info
+
+      - name: Create kyma-system namespace
+        run: |
+          make -C hack/ci/ create-kyma-system-ns
+
+      - name: Install the latest released NATS manager
+        run: |
+          kubectl apply -f https://github.com/kyma-project/nats-manager/releases/latest/download/nats-manager.yaml
+          kubectl apply -f https://github.com/kyma-project/nats-manager/releases/latest/download/nats-default-cr.yaml
+
+      - name: Wait for NATS module to be ready
+        run: |
+          make -C hack/ci/ wait-nats-cr-ready
+
+      - name: Install Eventing Manager before Upgrade
+        run: |
+          pre_upgrade_image=$(./hack/ci/build-full-image-path.sh ${{ inputs.pre-upgrade-image-tag }})
+          echo "Pre-upgrade image: $pre_upgrade_image"
+          make deploy IMG=$pre_upgrade_image
+
+      - name: Create test resources and waits for eventing CR readiness
+        run: |
+          export MANAGER_IMAGE=$(./hack/ci/build-full-image-path.sh ${{ inputs.pre-upgrade-image-tag }})
+          make e2e-setup
+          make e2e-eventing-setup
+
+      - name: Wait for the ${{ inputs.build-job-name }} job to succeed
+        if: ${{ inputs.build-job-name != '' }}
+        uses: kyma-project/wait-for-commit-status-action@2b3ffe09af8b6f40e1213d5fb7f91a7bd41ffb20
+        with:
+          context: ${{ inputs.build-job-name }}
+          commit_ref: "${{ inputs.build-job-commit-sha }}" # Note: 'github.event.pull_request.head.sha' is not same as 'github.sha' on pull requests.
```

Comment:
```suggestion
          commit_ref: "${{ inputs.build-job-commit-sha }}" # Note: 'github.event.pull_request.head.sha' is not the same as 'github.sha' on pull requests.
```

Reply:
yes, that's why I made it variable so that inputs.build-job-name can be set either PR commit sha or commit sha